### ISSUE TITLE
Support for client-go >=0.18.0

### ISSUE THREAD

**Issue by hmlkao** [hiddeco/cronjobber#43]
There is breaking change in [client-go v0.18.0](https://github.com/kubernetes/client-go/commit/ae9f6b2601c8f8b97ad2865943f423d65539ffdb) when is context required for REST client and is not possible to use generated clientset because context is missing (eg. in [Get() function](https://github.com/hiddeco/cronjobber/blob/4ae4a73bb196117b2e363edd9bd2f2fbdd460e2e/pkg/client/clientset/versioned/typed/cronjobber/v1alpha1/tzcronjob.go#L74)).

**DoD**:
* cronjobber clientset can be used with client-go version >= 0.18.0

**Comment by hiddeco** [comment]
You are absolutely right! This project is (sadly) suffering from my other (day job) OSS work.

If I would have a chance to redo this project, I would most likely write it from the ground up using controller-runtime while keeping the API specification the same. Reason is that there have been reports that the current core controller could be more efficient, and it increases long-term maintainability.

In addition, I would keep the API free from a typed client, and instead tell consumers to add the scheme to their dynamic client.

Getting this into a long-term maintainable shape again is something I am not likely going to do anytime soon, but I am open to review any contributions, including more complex ones. If someone wants to take on the task of maintaining this, you can reach out to me too. Either by commenting here, or sending me a message at `hello - at - hidde - dot - co`.

**Comment by jalonsoa** [comment]
Hi !
I have opened a pull request to update the kubernetes libraries and the associated go version (go 1.18 and kubernetes 0.24).

We have to review the tests, since they were executed with the image of circleci/golang and there is no image for version 1.18 of go.
